[WIP] Release notes for 7.3.0 (#146)
migtools/mta-documentation: docs/topics/ref_new-features-7-3-0.adoc

:_newdoc-version: 2.18.3
:_template-generated: 2025-04-17

:_mod-docs-content-type: REFERENCE

[id="new-features-7-3-0_{context}"]
= New features

This section provides the new features and improvements of the {ProductFullName} 7.3.0. 

MTA-4540

MTA-4549

> MTA-4610

Author: ```suggestion
MTA-4610

.Generate deployment manifest in {ProductShortName} CLI

Starting with {ProductShortName} {ProductVersion}, you can run the `discover` and `generate` commands in the {CLIName}. The `discover` command generates the discover manifest that preserves the platform and runtime configurations defined in the Cloud Foundry (CF) application manifest. The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format so that, it is ready for {ocp-full} deployments.

.{ProductShortName} uses Red Hat Build of Keycloak for user authentication

The {ProductShortName} {ProductVersion} uses `{rhbk-first}` instance for user authentication and authorization.The {ProductShortName} operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`.

.List all languages in an application

In {ProductShortName} {ProductVersion}, you can list all languages in an application using the `--list-languages` flag in the {CLIName}. The listed language that has no supported providers requires a custom rule set and the --`override-provider-settings` flag when you analyze the application.
```

:_newdoc-version: 2.18.3
:_template-generated: 2025-04-17

:_mod-docs-content-type: REFERENCE

[id="new-features-7-3-0_{context}"]
= New features

This section provides the new features and improvements of the {ProductFullName} 7.3.0. 

> .Generate deployment manifest in {ProductShortName} CLI

Author: ```suggestion
.The deployment manifest generation is now available in {ProductShortName} CLI
```

> .Asset generation is now available in {ProductShortName} CLI

Reviewer: ```suggestion
.Asset generation is available as Developer Preview in {ProductShortName} CLI
```

> .The deployment manifest generation is now available in {ProductShortName} CLI

Reviewer 2: This is for the asset generation feature

Author: @Pkylas007, please, check if anything needs to be clarified here :-)

Reviewer: @eemcmullan Can you please clarify if the title must change to:
```suggestion
.Asset generation is now available in {ProductShortName} CLI
```

Reviewer 2: LGTM

Starting with {ProductShortName} {ProductVersion}, you can run the `discover` and `generate` commands in the {CLIName}: 

* The `discover` command generates the discover manifest that preserves the platform and runtime configurations defined in the Cloud Foundry (CF) application manifest.

> * The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format to prepare it for {ocp-full} deployments.

Reviewer: ```suggestion
* The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format to prepare it for {ocp-full} deployments. You can use the command to  generate non-Kubernetes manifests such as a Dockerfile.
```

(link:https://issues.redhat.com/browse/MTA-4610[MTA-4610])

.{ProductShortName} now uses Red Hat Build of Keycloak for user authentication

{ProductShortName} {ProductVersion} now uses `{rhbk-first}` instance for user authentication and authorization. The {ProductShortName} Operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`. (link:https://issues.redhat.com/browse/MTA-4540[MTA-4540])

> .Listing all languages in an application is now supported

Author: @Pkylas007, could you please add the ticket number to this RN? 
Thanks :-)

> .Listing all languages in an application is now supported

Reviewer: ```suggestion
.Listing all languages in an application is  available as a Developer Preview
```

:_newdoc-version: 2.18.3
:_template-generated: 2025-04-17

:_mod-docs-content-type: REFERENCE

[id="new-features-7-3-0_{context}"]
= New features

This section provides the new features and improvements of the {ProductFullName} 7.3.0. 

> MTA-4540
> 
> 
> MTA-4549
> 
> MTA-4610

Author: ```suggestion
```

> .Generate deployment manifest in {ProductShortName} CLI

Author: How about this wording to emphasize it is a new feature? I know that the section's name implies that, I'm just following our RHEL standards, hehe :-D

```suggestion
.The deployment manifest generation is now supported in the {ProductShortName} CLI
```

Reviewer: I would prefer to retain the actionable title (Generate...) if that's fine.

Author: Hmm...I'd say that this kind of "actionable" title does suit some procedural module rather than a release note. A release note is a theoretical pice of info and that's why I'd prefer not using ing here especially because it sounds like an imperative and the user must do something, which is not the case.

Author: As for you suggestion ("Support to generate deployment manifest"), if you'd like to have something like that, I suggest rewording it a bit to "Support for deployment manifest generation in {ProductShortName} CLI".

Reviewer: @mpershina Shall we please agree to:
 
```suggestion
.Support for generating deployment manifest in {ProductShortName} CLI
```
Reason: titles must not be a complete sentence.

Author: Hmm, that's interesting...I usually follow our CCS Guidelines, and there's nothing like that about that: https://source.redhat.com/groups/public/ccs/rheldocslandingpage/rhel_documentation_master_page_wiki/rhel_release_notes_ccs_guidelines_for_writers#jive_content_id_12_General_rules_for_titles.

> Starting with {ProductShortName} {ProductVersion}, you can run the `discover` and `generate` commands in the {CLIName}. The `discover` command generates the discover manifest that preserves the platform and runtime configurations defined in the Cloud Foundry (CF) application manifest. The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format so that, it is ready for {ocp-full} deployments.

Author: If the case is in the word "run", can be. I don't have a strong preference here :-) 
As for "so that, it is ready for {ocp-full} deployments" , not sure about this "so that it is ready" part. I think "to prepare" here means the same. But if you decide to use "so that it is ready", don't forget to remove the comma :-)

Let me know if you meant something else by your suggestion :-D

> Starting with {ProductShortName} {ProductVersion}, you can run the `discover` and `generate` commands in the {CLIName}. The `discover` command generates the discover manifest that preserves the platform and runtime configurations defined in the Cloud Foundry (CF) application manifest. The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format so that, it is ready for {ocp-full} deployments.

Reviewer: Oh, yeah one of the changes was an omission. I meant keeping run the commands :)

> Starting with {ProductShortName} {ProductVersion}, you can run the `discover` and `generate` commands in the {CLIName}. The `discover` command generates the discover manifest that preserves the platform and runtime configurations defined in the Cloud Foundry (CF) application manifest. The `generate` command uses the discover manifest to generate the deployment manifest in a Kubernetes-native format so that, it is ready for {ocp-full} deployments.

Author: True :-) We just try to avoid conversational tone :-D

> .{ProductShortName} uses Red Hat Build of Keycloak for user authentication

Author: ```suggestion
.{ProductShortName} now uses Red Hat Build of Keycloak for user authentication
```

Reviewer: Ok

> The {ProductShortName} {ProductVersion} uses `{rhbk-first}` instance for user authentication and authorization.The {ProductShortName} operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`.

Author: One Q to this: You mentioned that the pod name for {rhbk-short} has been changed to `rhbk-operator`. What about  the secret then?

```suggestion
The {ProductShortName} {ProductVersion} now uses `{rhbk-first}` instance for user authentication and authorization. The {ProductShortName} Operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`.
```

Reviewer: I'm cannot apply the changes directly in this PR :) Yes, let's go with the changes please.

Author: Sure, applying :)

> {ProductShortName} {ProductVersion} now uses `{rhbk-first}` instance for user authentication and authorization. The {ProductShortName} Operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`.

Author: ```suggestion
{ProductShortName} {ProductVersion} now uses `{rhbk-first}` instance for user authentication and authorization. The {ProductShortName} Operator manages the `{rhbk-short}` instance and configures a dedicated realm with necessary roles and permissions. For `{rhbk-short}`, the secret and the pod name has changed. The pod name for {rhbk-short} is changed to `rhbk-operator`. (link:https://issues.redhat.com/browse/MTA-4540[MTA-4540])
```

> .List all languages in an application

Reviewer: I would prefer List since it's an action verb here than Listing that doesn't directly indicate that the user does something. What do you think?

> .List all languages in an application

Reviewer: > I would prefer List since it's an action verb in the title. What do you think?

```suggestion
.List all languages in an application
```

> .List all languages in an application

Reviewer: ```suggestion
. Support for listing languages in an application
```
Shall we please not enter full sentences in headings/titles?

Reviewer: i have reworked the title though I'm still for actionable titles :) Those speak directly to what the customers can do instead of what we present to them.

Author: Again, I have never heard about not using full sentences in headings/titles. See my comment on the similar case above

> In {ProductShortName} {ProductVersion}, you can list all languages in an application using the `--list-languages` flag in the {CLIName}. The listed language that has no supported providers requires a custom rule set and the --`override-provider-settings` flag when you analyze the application.

Author: ```suggestion
In {ProductShortName} {ProductVersion}, you can list all languages in an application by using the `--list-languages` flag in the {CLIName}. The listed language that has no supported providers requires a custom rule set and the --`override-provider-settings` flag when you analyze the application.
```

Reviewer: Sure, thanks! Could you apply the change please?
